### ISSUE TITLE
Visualize OncoKB annotation and generate sample/patient report

### ISSUE THREAD

**Issue by zhx828** [cBioPortal/GSoC#95]
**Background**: 
OncoKB™(https://www.oncokb.org) is a precision oncology knowledge base developed at Memorial Sloan Kettering Cancer Center that contains biological and clinical information about genomic alterations in cancer. cBioPortal includes OncoKB annotation in it's result view, patient view to provide detailed biological effect and clinical implications for each variant. 

OncoKB provides varies way for users to annotate alterations. We have standard APIs that can be used with high volume requests (https://api.oncokb.org). We also have a python package which builds on top the APIs to annotate MAF/CopyNumberAlteration/StructuralVariant. Even with these applications, portion of OncoKB users still have difficulty to get their variants annotated. Especially to the users that without any computational background.

**Goal**: 

1. Create an standard alone npm package that can be imported as dependency to visualize sample/patient level OncoKB annotation. The tool should include the following components.

    - Component to visualize OncoKB annotation by mutation type
    - Download the annotation as a PDF report

2. Integrate with cBioPortal sample view
    - Convert all sample genomic data into OncoKB API format
    - Annotate genomic data through OncoKB API
    - Use the component created in step 1 to visualize

### Mockup(subject to change)
![Screen Shot 2023-02-07 at 3 06 23 PM](https://user-images.githubusercontent.com/5400599/217366296-b756ba19-60df-4cf8-a777-2f5468b87291.png)

**Approach**: 
- Understand the project especially understand the input formats
- Work with team members to implement the components above

**Need skills**: 
- Javascript, typescript, React

**Possible mentors**: 
John Konecny, Hongxin Zhang(@zhx828 )
Mary Chapman @mlizchap

**Comment by Abinash-bit** [comment]
@zhx828 Hey... I am intrested in resolving this issue... Please guide me .

**Comment by chaitravi-ce** [comment]
@zhx828 @calvinlu3 I am deeply interested in contributing to this issue...Could you please guide me in the further steps I should follow

**Comment by tamal78** [comment]
Hi @zhx828 can i would like contribute to this project .I have gone through the codebase and all the guidelines.I am more than happy to contribute for CbioPortal for this project.Hoping for a reply from the community!

**Comment by zhx828** [comment]
Hi @Abinash-bit @chaitravi-ce @tamal78 , thank you for your interests. To create a successful proposal, I think you would need to do the following
- Get familiar with [oncokb-public website](https://www.oncokb.org/) and [code base](https://github.com/oncokb/oncokb-public)
- Understand what we do/not offer through API, api.oncokb.org
- Have a good understanding the data types, Mutation/Copy Number Alteration/Structural Variant
- Understand why we want to create the tool and find a valid use case
- (optional) grab a good first issue to work on. Please also comment on the ticket if you are actually working on it to avoid uncessary duplications. Feel free to reach out to me if you need clarification. https://github.com/oncokb/oncokb/labels/good%20first%20issue

**Comment by SalihuDickson** [comment]
Hello @zhx828. This project requires that you be familiar with the oncokb code base. And the description says you'll have to build a tool that will be integrated with the website. But the oncokb platform is built largely with java. Do we need to know java to work on this project, I know the listed requirements are javascript, typescript and react but can I please get a clarification on if java will be needed at all.

**Comment by zhx828** [comment]
> Hello @zhx828. This project requires that you be familiar with the oncokb code base. And the description says you'll have to build a tool that will be integrated with the website. But the oncokb platform is built largely with java. Do we need to know java to work on this project, I know the listed requirements are javascript, typescript and react but can I please get a clarification on if java will be needed at all.

@SalihuDickson This project would only require frontend coding. Any data if you need through API, our team will help on that.

**Comment by SalihuDickson** [comment]
@zhx828 Okay that sounds great, thank you. I'm really looking forward to having the opportunity to work on this project.

**Comment by zhx828** [comment]
@shrikant1809 thank you for your interest! It's great to hear our thoughts are resonated with other community members. If I understand correctly, you are a full time employee now at Medprime Technologies? This project is designed for GSoC participants and mostly students. But if I misunderstand anything, please let me know.

**Comment by Ahmed89shaaban** [comment]
I'm interested to begin this project and have new ideas  need to guide and discuss plz

**Comment by Beking0912** [comment]
Hi, @zhx828 ,
I'm Gloria, a first-year CS Master student at North Carolina State University, and I'm interested in this project. 
With around two year of experience worked as a Front-end Engineer writing React application with JavaScript and TypeScript, I've found that my tech stack is a great match for cBioPortal's front-end codebase. As a newcomer to open-source projects, I am eager to have the opportunity to make a contribution. 
Please let me know if this project has already been assigned. Thank you for your time.

**Comment by shrikant1809** [comment]
> @shrikant1809 thank you for your interest! It's great to hear our thoughts are resonated with other community members. If I understand correctly, you are a full time employee now at Medprime Technologies? This project is designed for GSoC participants and mostly students. But if I misunderstand anything, please let me know.

Yes, I am a full time employee, but I can come up with arrangements for participating in GSoC and wish to contribute to community.
I hope you will consider my request.

**Comment by Ahmed89shaaban** [comment]
We cand collebrate to make good discussion and find good idea

**Comment by divz-codess** [comment]
Hi I am interested in working on this project under GSoC 2023. I am currently in my junior year of university and have about an year of experience working as a frontend developer in past internships using various tech stacks such as ReactJS, Flutter etc. Additionally I have a research paper published in the field of Biological sciences and have deep interest in Genomics and Healthcare. Therefore, I will be really grateful for any help and guidance for this project. 
Looking forward to your reply @zhx828  
Have a nice day :)

**Comment by Ahmed89shaaban** [comment]
Hi  iam very happy to contribute with you and discuss all idea and to be
team

**Comment by zhx828** [comment]
@Ahmed89shaaban @Beking0912 @shrikant1809 @divz-codess thank you all for your interests. The description on this project should have the info to get you started. When ready, feel free to [slack me ](https://slack.cbioportal.org/) your proposal before submitting. I can provide some feedbacks. Please have it uploaded to google doc and enable the access. Thank you.

**Comment by shrikant1809** [comment]
Greetings @zhx828,
Hope you find this message well. As instructed I have written a proposal for GSoC 2023. I have included Project titles, vision, detailed approach and my contact information. I have create proposal similar to the template which was available on GitHub of cBioPortal and sent you on slack. I hope you will approve my request to contribute and give a favourable response and approve my candidature.
Thank you.

**Comment by Shashank2104125** [comment]
@zhx828 I am really interested in this project. Will this project open for GSoC 2024?

**Comment by zhx828** [comment]
> @zhx828 I am really interested in this project. Will this project open for GSoC 2024?

Yes it will. Sorry about the late reply.

**Comment by heymitali** [comment]
Hello, my name is Mitali. I have a strong interest in this project and am eagerly anticipating the opportunity to contribute to it.
